### PR TITLE
feat: refactor block, remove the data enum and type enum

### PR DESCRIPTION
**Pull request by LucasXu0**
- [x] Remove the data enum and type enum.
- [x] Add tests for block.rs.
- [x] Fix the missing logic in document.rs.
- [x] Add external id and type.

### CODE REVIEW

**Comment by LucasXu0** [comment]
For example,
paragraph block
```
{
    "type": "paragraph",
    "data": {},
    "extend_id": "3d3d0088-ab9a-4e9d-9db9-31078fd6e4c8",
    "extend_type": "y_text"
}
```

list block
```
{
    "type": "bulleted_list",
    "data": {},
    "children": [
        {
            "type": "paragraph",
            "data": {},
            "extend_id": "3d3d0088-ab9a-4e9d-9db9-31078fd6e4c8",
            "extend_type": "y_text",
            "children": [
                {
                    "type": "paragraph",
                    "data": {},
                    "extend_id": "3d3d0088-ab9a-4e9d-9db9-31078fd6e4c9",
                    "extend_type": "y_text"
                }
            ]
        }
    ]
}
```
* ABC
  123

File: collab-document/src/document.rs [hunk 1]
```diff
@@ -14,11 +18,10 @@ const CHILDREN_MAP: &str = "children_map";
 pub struct Document {
   #[allow(dead_code)]
   inner: Collab,
-  root: MapRefWrapper,
-  text_map: TextMap,
-  children_map: ChildrenMap,
+  pub root: MapRefWrapper,
+  pub text_map: TextMap,
+  pub children_map: ChildrenMap,
   pub blocks: BlockMap,
```
Comment (LucasXu0) [inline collab-document/src/document.rs:24]: @appflowy I want to use  `trait` to define the blocks, like

```rs
pub trait OperableBlocks {}
// ...
pub blocks: OperableBlocks
```

However, I encountered a compiler error when trying to use Box<dyn OperableBlocks>. Do you have any suggestions for how to fix it?

File: collab-document/src/blocks/block.rs [hunk 2]
```diff
@@ -1,163 +1,172 @@
...
-    };
-    s.serialize_field("data", &data)?;
-
-    s.end()
-  }
-}
-pub struct BlockMap {
-  root: MapRefWrapper,
+pub struct BlockOperation {
+  pub root: MapRefWrapper,
```
Comment (appflowy) [inline collab-document/src/blocks/block.rs:33]: private?

File: collab-document/tests/delta/util.rs [hunk 3]
```diff
@@ -28,38 +30,65 @@ pub fn create_document(doc_id: &str) -> DocumentTest {
...
-    document.move_block(txn, block_id, parent_id, prev_id);
-  });
+pub fn move_block(
+  document: &Document,
+  block_id: &str,
+  parent_id: &str,
+  prev_id: &str,
+) -> Result<(), DocumentError> {
+  document
+    .root
```
Comment (appflowy) [inline collab-document/tests/delta/util.rs:90]: try hide the root. You can` impl Deref for Document` also `DerefMut`

**Pull request merged**